webroot/src/pages/FinalizePrivilegePurchase/FinalizePrivilegePurchase.ts

```diff
@@ -0,0 +1,528 @@
+                stateMilitaryPurchaseText
+            };
+        });
+    }
+
+    get currentCompact(): Compact | null {
+        return this.userStore?.currentCompact || null;
+    }
+
+    get currentCompactType(): string | null {
+        return this.currentCompact?.type || null;
+    }
+
+    get currentCompactCommissionFee(): number | null {
+        return this.currentCompact?.compactCommissionFee || null;
+    }
+
+    get stateFeeText(): string {
+        return this.$t('payment.expirationDate');
+    }
+
+    get arePurchaseAttestationsAccepted(): boolean {
+        return this.userStore.arePurchaseAttestationsAccepted;
+    }
+
+    get purchaseDataList(): Array<PrivilegePurchaseOption> {
+        return this.currentCompact?.privilegePurchaseOptions || [];
+    }
+
+    get compactPrivilegeStateFeeText(): string {
+        return this.$t('payment.compactPrivilegeStateFee');
+    }
+
+    get compactCommissionFeeText(): string | null {
+        return `${this.$t('payment.compactCommissionFee')} ($${this.currentCompactCommissionFee?.toFixed(2)} x ${this.privCount})`;
+    }
+
+    get privCount(): number {
+        return this.selectedStatePurchaseDataList?.length || 0;
+    }
+
+    get totalTitle(): string {
+        return this.$t('common.total');
+    }
+
+    get totalCompactCommissionFee(): number {
+        let total = 0;
+
+        if (this.currentCompactCommissionFee) {
+            total = this.privCount * this.currentCompactCommissionFee;
+        }
+
+        return total;
+    }
+
+    get militaryDiscountText(): string {
+        return this.$t('licensing.militaryDiscountText');
+    }
+
+    get formValidationErrorMessage(): string {
+        return this.$t('common.formValidationErrorMessage');
+    }
+
+    get totalPurchasePrice(): number {
+        let total = this.totalCompactCommissionFee;
+
+        this.selectedStatePurchaseDataList.forEach((stateSelected) => {
+            if (stateSelected?.fee) {
+                total += stateSelected.fee;
+            }
+
+            if (stateSelected?.isMilitaryDiscountActive && stateSelected?.militaryDiscountAmount) {
+                total -= stateSelected.militaryDiscountAmount;
+            }
+        });
+
+        return total;
+    }
+
+    get isSubmitEnabled(): boolean {
+        this.checkValidForAll();
+
+        return this.isFormValid && this.formData.noRefunds.value && !this.isFormLoading;
+    }
+
+    //
+    // Methods
+    //
+    async handleSubmit() {
+        this.validateAll({ asTouched: true });
+
+        if (this.isFormValid) {
+            this.startFormLoading();
+            this.isFormError = false;
+            this.formErrorMessage = '';
+
+            const { formValues, statesSelected } = this;
+
+            const serverData = LicenseeUserPurchaseSerializer.toServer({ formValues, statesSelected });
+
+            const purchaseServerEvent = await this.$store.dispatch('user/postPrivilegePurchases', serverData);
```

Comment:
Probably don't need these empty lines between variable declarations.

Also, minor, but I've been putting the `handleSubmit()` methods lower down than the `initFormInputs()` methods.